File: Python/Easy/3432.py
from typing import List 
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total = sum(nums)
        accumulate = 0
        count = 0
        for i in range(len(nums) - 1):
            accumulate += nums[i]
            total -= nums[i]
            if abs(accumulate - total) % 2 == 0:
                count += 1
        return count 
    

File: C++/Easy/3432.cc
#include <numeric>
#include <vector>
#include <cmath>


using namespace std;
class Solution {
public:
    int countPartitions(vector<int>& nums) {
        int subtotal{0}, count{0};
        int total = accumulate(nums.begin(), nums.end(), 0);
        for (int i = 0; i < nums.size() - 1; ++i){
            subtotal += nums[i];
            total -= nums[i];
            if (abs(subtotal - total) % 2 == 0){
                count += 1;
            }
        }
        return count;
    }
};